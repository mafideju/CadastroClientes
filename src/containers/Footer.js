import React from 'react';

const Footer = () => (
  <footer className="footer">
    <div className="footer__content">
      <div className="content-container">
        <h4 className="footer__content--text">
          Desenvolvido em 2018 pela &copy;MAFFEE usando React / Redux
        </h4>
      </div>
    </div>
  </footer>
);

export default Footer;
