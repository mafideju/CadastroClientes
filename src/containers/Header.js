import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
  <header className="header">
    <div style={{ margin: '0 1rem' }}>
      <div className="header__content">
        <NavLink className="header__title" exact to="/dashboard">
          <h1 className="header__title--animated-left">Cadastro Clientes</h1>
        </NavLink>
        {/*
    <NavLink to="/create" activeClassName="orange">
      Criar
    </NavLink>
    <NavLink to="/help" activeClassName="orange">
      Ajuda
    </NavLink>
    */}
        <button
          className="button button--logout button--animated__right"
          onClick={startLogout}
        >
          Sair
        </button>
      </div>
    </div>
  </header>
);

const mapDispatchToProps = dispatch => ({
  startLogout: () => dispatch(startLogout())
});

export default connect(
  undefined,
  mapDispatchToProps
)(Header);
