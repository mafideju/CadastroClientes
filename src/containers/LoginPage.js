import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export const LoginPage = ({ startLogin }) => (
  <div className="box-layout">
    <div className="box-layout__box">
      <h1 className="box-layout__title">**Boilerplate**</h1>
      <p>Faça as Edições Necessárias</p>
      <button
        className="button button--animated button--login"
        onClick={startLogin}
      >
        Google Login
      </button>
      <button className="button button--animated button--login">
        E-mail Login
      </button>
    </div>
  </div>
);

const mapDispatchToProps = dispatch => ({
  startLogin: () => dispatch(startLogin())
});

export default connect(
  undefined,
  mapDispatchToProps
)(LoginPage);
