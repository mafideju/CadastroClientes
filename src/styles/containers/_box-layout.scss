.box-layout {
  align-items: center;
  background: url('/images/bgcoins100.jpg');
  background-size: cover;
  display: flex;
  height: 100vh;
  justify-content: center;
}

.box-layout__box {
  background: fade-out($orange-one, 0.25);
  border-radius: 10px;
  padding: $lg-size $md-size;
  text-align: center;
  width: 25rem;
}

.box-layout__title {
  margin: 0 0 $md-size 0;
  line-height: 1.4;
  animation-name: moveInTop;
  animation-duration: 1.5s;
  animation-delay: 0.5s;
  animation-timing-function: ease-out;
}
@keyframes moveInTop {
  0% {
    opacity: 0;
    transform: translateY(-15rem);
  }

  80% {
    transform: translateY(1rem);
  }

  100% {
    opacity: 1;
    transform: translate(0);
  }
}
