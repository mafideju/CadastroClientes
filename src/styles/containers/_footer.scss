.footer {
  background-image: linear-gradient(
    to right bottom,
    lighten($orange-three, 15%),
    darken($orange-three, 10%)
  );

  &__content {
    align-items: center;
    display: flex;
    justify-content: space-between;
    padding: $sm-size 0;

    &--text {
      color: white;
      font-size: $md-size;
      text-align: center;
    }
  }
}
