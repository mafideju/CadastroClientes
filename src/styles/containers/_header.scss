.header {
  background-image: linear-gradient(
    to right bottom,
    lighten($orange-three, 15%),
    darken($orange-three, 10%)
  );

  &__title {
    color: white;
    text-decoration: none;

    &,
    h1 {
      margin: 0;
    }
  }

  &__title--animated-left {
    animation-name: moveInLeft;
    animation-duration: 1.5s;
    animation-delay: 0.5s;
    animation-timing-function: ease-out;
  }

  &__content {
    align-items: center;
    display: flex;
    justify-content: space-between;
    padding: $sm-size 0;
  }
}
@keyframes moveInLeft {
  0% {
    opacity: 0;
    transform: translateX(-15rem);
  }

  80% {
    transform: translateX(1rem);
  }

  100% {
    opacity: 1;
    transform: translate(0);
  }
}
