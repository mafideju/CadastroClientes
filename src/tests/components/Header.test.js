import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../containers/Header';

test('HEADER COMPONENT TESTING', () => {
  const wrapper = shallow(<Header startLogout={() => {}} />);
  expect(wrapper).toMatchSnapshot();
});

test('LOGOUT NO CLICK DO BOTÃƒO', () => {
  const startLogout = jest.fn();
  const wrapper = shallow(<Header startLogout={startLogout} />);
  wrapper.find('button').simulate('click');
  expect(startLogout).toHaveBeenCalled();
});
