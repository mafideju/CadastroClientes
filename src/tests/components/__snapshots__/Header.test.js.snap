// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HEADER COMPONENT TESTING 1`] = `
<header
  className="header"
>
  <div
    className="content-container"
  >
    <div
      className="header__content"
    >
      <NavLink
        activeClassName="active"
        aria-current="page"
        className="header__title"
        exact={true}
        to="/dashboard"
      >
        <h1
          className="header__title--animated-left"
        >
          **Boilerplate**
        </h1>
      </NavLink>
      <button
        className="button button--logout button--animated__right"
        onClick={[Function]}
      >
        Sair
      </button>
    </div>
  </div>
</header>
`;
